x-pack/plugins/fleet/server/services/agents/unenroll_action_runner.ts

```diff
@@ -126,6 +138,82 @@ export async function unenrollBatch(
   };
 }
 
+export async function updateActionsForForceUnenroll(
+  esClient: ElasticsearchClient,
+  agentIds: string[],
+  actionId: string,
+  total: number
+) {
+  // creating an unenroll so that force unenroll shows up in activity
+  await createAgentAction(esClient, {
+    id: actionId,
+    agents: [],
+    created_at: new Date().toISOString(),
+    type: 'FORCE_UNENROLL',
+    total,
+  });
+  await bulkCreateAgentActionResults(
+    esClient,
+    agentIds.map((agentId) => ({
+      agentId,
+      actionId,
+    }))
+  );
+
+  // updating action results for those agents that are there in a pending unenroll action
+  const unenrollActions = await getUnenrollAgentActions(esClient);
+  for (const action of unenrollActions) {
+    const commonAgents = intersection(action.agents, agentIds);
+    if (commonAgents.length > 0) {
+      // filtering out agents with action results
+      const agentsToUpdate = await getAgentsWithoutActionResults(
```

Comment:
Added one more check to only update action results for those agents that do not yet have action results. This is to avoid having multiple result entries for the same action.
Did some measurements on 100k agent entries, and the unenroll took 24s without revoke and 31s with revoke.